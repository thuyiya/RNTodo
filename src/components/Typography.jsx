import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';

import { theme } from '../constants'

const Typography = ({
    size = 'h5', 
    weight= 'normal', 
    space = 0,
    align = 'center',
    style = {},
    color = theme.colors.textColor,
    children
}) => {
    return (<Text style={[{
        margin: space * theme.spaceSize,
        textAlign: align,
        color: color,
        fontSize: theme.fontSizeMap[size],
        fontWeight: theme.fontWeightMap[weight]
    }, style]}>{children}</Text>)
}

Typography.propTypes = {
    size: PropTypes.oneOf([...Object.keys(theme.fontSizeMap)]),
    weight: PropTypes.oneOf([...Object.keys(theme.fontWeightMap)]),
    style: PropTypes.object
}

export default Typography;