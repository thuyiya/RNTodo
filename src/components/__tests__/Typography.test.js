import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react-native";
import Typography from "../Typography";

describe("<LoadingView />", () => {
  const rendetText = "Hello";
  it("has 1 children", () => {
    const tree = renderer
      .create(<Typography>{rendetText}</Typography>)
      .toJSON();
    expect(tree.children.length).toBe(1);
  });

  it("no children text", () => {
    const tree = renderer
      .create(<Typography></Typography>)
      .toJSON();
    expect(tree.children).toBe(null);
  });

    it(`displays the test as ${rendetText}`, () => {
      render(<Typography>{rendetText}</Typography>);
      expect(screen.queryByText(rendetText)).toBeTruthy();
    });

  it("renders correctly", () => {
    const tree = renderer
      .create(<Typography>{rendetText}</Typography>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
