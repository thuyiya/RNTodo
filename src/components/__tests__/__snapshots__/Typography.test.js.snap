// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingView /> renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#333333",
        "fontSize": 16,
        "fontWeight": "300",
        "margin": 0,
        "textAlign": "center",
      },
      {},
    ]
  }
>
  Hello
</Text>
`;
